lib/Twig/SourceContextLoaderInterface.php

```diff
@@ -9,6 +9,13 @@
  * file that was distributed with this source code.
  */
 
+/**
+ * Adds a getSourceContext() method for loaders.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @deprecated since 1.27 (to be removed in 3.0)
```

Comment:
This deprecation prevents a clean upgrade path, doesn't it? You now need to implement a deprecated interface (`Twig_SourceContextLoaderInterface`) to get rid of a deprecated method (`Twig_LoaderInterface::getSource()`).